src/actions/attack.cpp

```diff
@@ -1687,7 +1691,7 @@ std::pair<int, bool> ability_leadership(const std::string& ability,const unit_ma
 		} else if(up == units.end() && (filter_student || filter || filter_attacker || filter_defender)) {
 			return {abil_value, false};
 		} else {
-			show_result = !(!un->abilities_filter_matches(*i->first, attacker, abil_value) || ability_apply_filter(un, up, ability, *i->first, loc, opp_loc, attacker));
+			show_result = !(!un->abilities_filter_matches(*i->first, attacker, abil_value) || ability_apply_filter(un, up, ability, *i->first, loc, opp_loc, attacker, weapon, opp_weapon));
```

Comment:
ability_apply_filter in after un->get_abilities(ability, weapon, opp_weapon); [filter_weapon/filtr_second_weapon] will be used in all case or these filter will be in abilitie like weapon in same time what usual code syntax what i want emulate, impossible to create an priority to the last, the only solution is to erase [filter_weapon/second_weapon] directly. It is one or other but not both.